### DISCUSSION TITLE
Current spam attack: Mastodon is the problem

### DISCUSSION THREAD

**Discussion by todoneunl** [mastodon/mastodon#29267] [General]
After around 48 hours of spam, I noticed one thing during mitigating the current spam attack on my server; the spam accounts are all located on Mastodon servers. As far as I can see, the bot automatically creates accounts, also on fully updated Mastodon servers. The rumor goes that it's even a child or young teenager is doing this.

There are two sides of this spam attack:

1. It is very easy to automatically create Mastodon accounts with a script. Yes I know there options to block disposable email providers and IP's, but this shouldn't be easy at all. So please make this a high priority. Actually I hope it already is.

2. At the receiving side there is not much that we can do about the spam, except blocking affected servers, after the spam is already received. There are no tools to automatically mitigate this. I can say that this is very frustrating. So please, help us to stop this!? @Gargron send a post (after two days) how to make sure your server can't be hijacked, but those admins probably won't see this. So please make sure it is also a priority to create tools to stop the spam ASAP.

Please see these issues (two of them are created by me and are related):

*Require blocking of disposable email providers and/or require a captcha provider when registrations are open*

https://github.com/mastodon/mastodon/issues/29270

*Set new registrations on new servers to manual approval by default*

https://github.com/mastodon/mastodon/issues/29269

*Ability to greylist new servers*

https://github.com/mastodon/mastodon/issues/29266

*Ability to use heuristic spam filtering tools*

https://github.com/mastodon/mastodon/issues/29265

*Instance-wide filtering*

https://github.com/mastodon/mastodon/issues/29256

cc @ClearlyClaire @renchap

**Comment by RinCat** [comment]
>  the spam accounts are all located on Mastodon servers. As far as I can see, the bot automatically creates accounts, also on fully updated Mastodon servers. The rumor goes that it's even a child or young teenager is doing this.

The first wave of the spam are all coming from Misskey servers, as the attacker is targeting Japanese users. Later, as the admins take action, and there isn't many Misskey servers, the attacker moved to use Mastodon servers that allows public registration. 
 
 There's not much you can do without destroying the very idea of the federation system.

**Reply by todoneunl** [reply]
Of course you can do something. Make sure it's much more difficult to register accounts with bots. I have offered two ideas, but there are probably many more.

**Reply by RinCat** [reply]
You don't even need to register on exist servers. Get some very cheap or free domains can do the same thing.

**Reply by kikobar** [reply]
Not sure they would be able to do this with empty new cheap servers in exactly the same way, because they'll still need to get someone to follow their accounts in our servers for their servers to be discovered, won't they?

**Comment by mfa777** [comment]
Mastodon works like email system. But mastodon doesn't have any auto SPAM filter. SPAM is on the way, Mastodon need a junk box for each account.

**Comment by edoswald** [comment]
I've spent almost three hours cleaning up this mess. What this proves to me is the priorities of the devs are screwed up. We're arguing over icons, when I literally have 100GB of CRAP clogging my object storage, and growing. Eugen needs to wake up. We are getting royally slammed out here, and no one is doing jack s**t. 

This isn't any better than Twitter or X at this point.

**Comment by perryflynn** [comment]
It's quite easy to find the SPAM toots via API, my script is detecting them via the images and some japanese text and deletes/blocks the toots/senders. Luckily the messages are very simple and there are only a few different ones.

https://github.com/perryflynn/fediscripts

But this will become a huge problem in future SPAM waves...

**Comment by gnh1201** [comment]
After reviewing it, I developed a spam filter based on the [K-Anonymity](https://en.wikipedia.org/wiki/K-anonymity) theory. Currently, my server is no longer receiving any spam. :)

https://github.com/gnh1201/caterpillar

**Comment by todoneunl** [comment]
Closing this discussion, because the Mastodon developers are listening. 🎉 https://github.com/mastodon/mastodon/releases/tag/v4.2.8
